crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -866,6 +879,23 @@ where
 
         None
     }
+
+    /// Get the canonical header for the requested block. This checks the database for both the
+    /// block header and the block hash.
+    fn get_canonical_header(
+        &self,
+        block_number: BlockNumber,
+    ) -> Result<SealedHeader, reth_interfaces::Error> {
+        let max_hash = match self.blockchain.block_hash(block_number)? {
+            Some(block) => block,
+            None => return Err(Error::Provider(ProviderError::HeaderNotFound(block_number.into()))),
+        };
+        let max_header = match self.blockchain.header_by_number(block_number)? {
+            Some(block) => block,
+            None => return Err(Error::Provider(ProviderError::HeaderNotFound(block_number.into()))),
+        };
```

Comment:
this should be a provider function instead,

see #2769

Reply:
Added in a separate PR, #2801

I can add a followup to remove this method and use the provider method